32blit/graphics/surface.cpp

```diff
@@ -485,45 +537,65 @@ namespace blit {
    */
   void Surface::load_from_packed(const packed_image *image) {        
     uint8_t *palette_entries = ((uint8_t *)image) + sizeof(packed_image);
-    uint8_t *bytes = ((uint8_t *)image) + sizeof(packed_image) + (image->palette_entry_count * 4);
+
+    int palette_entry_count = image->palette_entry_count;
+    if(palette_entry_count == 0)
```

Comment:
Hmm, this should probably be `&& format == P` for non-paletted images to work... (Not that the tool can generate those)